crates/wasmparser/src/operators_validator.rs

```diff
@@ -702,28 +699,30 @@ impl OperatorValidator {
             }
             Operator::BrTable { ref table } => {
                 self.pop_operand(Some(Type::I32))?;
-                let mut label = None;
+                let default = self.jump(table.default())?;
+                let default_types = label_types(default.0, resources, default.1)?;
                 for element in table.targets() {
-                    let (relative_depth, _is_default) = element.map_err(|mut e| {
+                    let relative_depth = element.map_err(|mut e| {
                         e.inner.offset = usize::max_value();
                         OperatorValidatorError(e)
                     })?;
                     let block = self.jump(relative_depth)?;
-                    match label {
-                        None => label = Some(block),
-                        Some(prev) => {
-                            let a = label_types(block.0, resources, block.1)?;
-                            let b = label_types(prev.0, resources, prev.1)?;
-                            if a.ne(b) {
-                                bail_op_err!(
-                                    "type mismatch: br_table target labels have different types"
-                                );
-                            }
-                        }
+                    let tys = label_types(block.0, resources, block.1)?;
+                    if tys.len() != default_types.len() {
+                        bail_op_err!(
+                            "type mismatch: br_table target labels have different number of types"
+                        );
+                    }
+                    debug_assert!(self.br_table_tmp.len() == 0);
+                    for ty in tys.rev() {
+                        let ty = self.pop_operand(Some(ty))?;
+                        self.br_table_tmp.push(ty);
+                    }
+                    for ty in self.br_table_tmp.drain(..).rev() {
+                        self.operands.push(ty);
```

Comment:
Use `extend` here to help LLVM avoid do-I-need-to-grow checks? (Even though we know there will never need to be a grow because these types all just came from the stack).